src/revlanguage/datatypes/basic/Probability.cpp

```diff
@@ -23,19 +23,25 @@ Probability::Probability( void ) : RealPos( 1.0 )
 /** Construct from double */
 Probability::Probability( double x ) : RealPos( x )
 {
-    
-    if ( x < 0.0 || x > 1.0)
-    {
-        throw RbException() << "Creation of " << getClassType() << " with value x=" << x << " outside standard probabilities (0,1)";
-    }
-    
+    validate();
 }
 
 
 /** Construct from double */
 Probability::Probability( RevBayesCore::TypedDagNode<double> *x ) : RealPos( x )
 {
-    
+    validate();
+}
+
+void Probability::validate(double x) const
+{
+    if (x < 0 or x > 1)
+	throw RbException() << "Creation of " << getClassType() << " with value x=" << x << " outside standard probabilities [0,1]";
```

Comment:
I'm wondering if we can use `std::setprecision` to print more decimals for `x` than C++ does by default, because this can currently lead to somewhat confusing error messages:

```
> a <- Probability(1.000001)
   Error:	Creation of Probability with value x=1 outside standard probabilities [0,1]
```

Reply:
OK, I modified RbException so that std::setprecision now works with it.  Then I fixed the problem with the error message that you point out above.